components/script/script_runtime.rs

```diff
@@ -776,3 +780,44 @@ impl Deref for JSContext {
         &self.0
     }
 }
+
+pub struct StreamConsumer(*mut JSStreamConsumer);
+
+#[allow(unsafe_code)]
+impl StreamConsumer {
+    fn consumeChunk(sc: &mut JSStreamConsumer, begin: *const u8, length: usize) -> bool {
+        unsafe {
+            return StreamConsumerConsumeChunk(sc, begin, length);
+        }
+    }
+
+    fn streamEnd(sc: &mut JSStreamConsumer){
+        unsafe {
+            StreamConsumerStreamEnd(sc);
+        }
+    }
+
+    fn streamError(sc: &mut JSStreamConsumer, errorCode: usize){
```

Comment:
All of these methods should accept a `&self` argument instead of `sc: &mut JSStreamConsumer`, and pass `self.0` as the argument to the JSAPI function that is being called instead. This will allow us to use the objects like `some_stream_consumer.streamError(error_code)` from Rust code.

Reply:
Done